packages/core/src/runtime.ts

```diff
@@ -617,21 +617,18 @@ export class AgentRuntime implements IAgentRuntime {
             modelClass: ModelClass.SMALL,
         });
 
-        const parsedResult = parseJsonArrayFromText(
+        const evaluators = parseJsonArrayFromText(
             result
         ) as unknown as string[];
 
-        this.evaluators
-            .filter((evaluator: Evaluator) =>
-                parsedResult?.includes(evaluator.name)
-            )
-            .forEach((evaluator: Evaluator) => {
-                if (!evaluator?.handler) return;
+        for (const evaluator of this.evaluators) {
+            if (!evaluators.includes(evaluator.name)) continue;
 
-                evaluator.handler(this, message);
-            });
+            if (evaluator.handler)
+                await evaluator.handler(this, message, state, {}, callback);
```

Comment:
do we want this to callback multiple times?

Reply:
its only being called once per evaluator